geotrellis/geotrellis-contrib
# Publish gt-contrib artifacts automatically to SonaType via CircleCI

# Overview

Switch from manual Bintray publishing to automated CI-driven publishing to SonaType, using the same general workflow currently in use on https://github.com/geotrellis/geotrellis-server and https://github.com/geotrellis/maml

TODO: 
- [x] Hook up PR UI in GitHub to include a CircleCI "tests passing" check
- [x] Add a docker-compose container for the `benchmark` subproject in order to cleanup running benchmarks
- [ ] ~~Bump to Scala 2.12.10?~~

## Demo

<img width="1126" alt="Screen Shot 2019-09-25 at 1 48 35 PM" src="https://user-images.githubusercontent.com/1818302/65626496-83763d80-df9b-11e9-9020-14f697dfd68d.png">

## Notes

See discussion on original issue regarding the choice of the `machine` executor that just runs the new local docker-compose setup: https://github.com/geotrellis/geotrellis-contrib/issues/222#issuecomment-535041064

As part of this effort, I updated this repo to default to `develop` and protected both the `develop` and `master` branch. 

I removed the TravisCI webhooks for this repo from the repo settings in GitHub. I also turned off all toggles that I could get to from the TravisCI dashboard. I believe TravisCI is purged from this repo, but it would be worth double-checking.

Kudos to @rbreslow for constructing a straightforward, reusable workflow to follow for this effort.

The first publish task I attempted ran into a 503 Service Temporarily Unavailable error for sonatype. Not sure how transient those are...  https://circleci.com/gh/geotrellis/geotrellis-contrib/17

## Testing Instructions

`./scripts/test` should complete, running vlm, gdal and slick tests inside Docker via docker-compose. 

`./scripts/benchmark` should build and begin running. Once its running, feel free to kill, as on a typical laptop processor it will take a very long time (> 1hr) to complete.

The build for this branch should be green: https://circleci.com/gh/geotrellis/geotrellis-contrib/tree/feature%2Fawf%2Fci

Closes #222, Closes #122
